observatory-platform/observatory/platform/observatory_config.py

```diff
@@ -0,0 +1,1023 @@
+                 terraform: Terraform = None,
+                 airflow_variables: List[AirflowVariable] = None,
+                 airflow_connections: List[AirflowConnection] = None,
+                 dags_projects: List[DagsProject] = None,
+                 cloud_sql_database: CloudSqlDatabase = None,
+                 airflow_main_vm: VirtualMachine = None,
+                 airflow_worker_vm: VirtualMachine = None,
+                 validator: ObservatoryConfigValidator = None):
+        """ Create a TerraformConfig instance.
+
+        :param backend: the backend config.
+        :param airflow: the Airflow config.
+        :param google_cloud: the Google Cloud config.
+        :param terraform: the Terraform config.
+        :param airflow_variables: a list of Airflow variables.
+        :param airflow_connections: a list of Airflow connections.
+        :param dags_projects: a list of DAGs projects.
+        :param cloud_sql_database: a Google Cloud SQL database config.
+        :param airflow_main_vm: the Airflow Main VM config.
+        :param airflow_worker_vm: the Airflow Worker VM config.
+        :param validator: an ObservatoryConfigValidator instance.
+        """
+
+        super().__init__(backend=backend,
+                         airflow=airflow,
+                         google_cloud=google_cloud,
+                         terraform=terraform,
+                         airflow_variables=airflow_variables,
+                         airflow_connections=airflow_connections,
+                         dags_projects=dags_projects,
+                         validator=validator)
+        self.cloud_sql_database = cloud_sql_database
+        self.airflow_main_vm = airflow_main_vm
+        self.airflow_worker_vm = airflow_worker_vm
+
+    @property
+    def terraform_workspace_id(self):
+        """ The Terraform workspace id.
+
+        :return: the terraform workspace id.
+        """
+
+        return self.terraform.workspace_prefix + self.backend.environment.value
+
+    def terraform_variables(self) -> List[TerraformVariable]:
+        """ Create a list of TerraformVariable instances from the Terraform Config.
+
+        :return: a list of TerraformVariable instances.
+        """
+
+        sensitive = True
+        return [TerraformVariable('environment', self.backend.environment.value),
+                TerraformVariable('airflow', self.airflow.to_hcl(), sensitive=sensitive, hcl=True),
+                TerraformVariable('google_cloud', self.airflow.to_hcl(), sensitive=sensitive, hcl=True),
+                TerraformVariable('database', self.airflow.to_hcl(), sensitive=sensitive, hcl=True),
+                TerraformVariable('airflow_main_vm', self.airflow_main_vm.to_hcl(), hcl=True),
+                TerraformVariable('airflow_worker_vm', self.airflow_worker_vm.to_hcl(), hcl=True),
+                TerraformVariable('airflow_variables', list_to_hcl(self.airflow_variables), hcl=True,
```

Comment:
Yeah that's a good point, I can do that.

It would probably be good to be able to turn sensitive off from the command line for debugging purposes for all of the Terraform variables, but I can do that later.